platform_plugin_teams/api/lms/serializers.py

```diff
@@ -0,0 +1,38 @@
+"""Serializers for the Teams API."""
+from lms.djangoapps.teams.models import CourseTeam
+from lms.djangoapps.teams.serializers import BulkTeamCountTopicSerializer, CourseTeamSerializer
+from rest_framework import serializers
+
+
+class CustomTeamSerializer(BulkTeamCountTopicSerializer):
+    """
+    Serializer for add teams information to the topic.
+
+    This is a subclass of the BulkTeamCountTopicSerializer. The purpose of this
+    subclass is add the `teams` field, which is a list of teams.
+    """
+
+    teams = serializers.SerializerMethodField()
+
+    def get_teams(self, topic: dict) -> list:
+        """
+        Get the teams for a given topic.
+
+        Args:
+            topic (dict): The topic for which to get the teams.
+
+        Returns:
+            list: A list of teams for the given topic.
+        """
+        result_filter = self.context["result_filter"]
+        excluded_private_team_ids = self.context["excluded_private_team_ids"]
+
+        result_filter.update({"topic_id": topic.get("id")})
+        queryset = CourseTeam.objects.filter(**result_filter).exclude(
```

Comment:
same comment here about using the model instead of an API method

Reply:
I was searching and did not find a method to get a team by custom filter